content/docs/02-foundations/04-tools.mdx

```diff
@@ -8,65 +8,43 @@ description: Learn about tools with the Vercel AI SDK.
 While [large language models (LLMs)](/docs/foundations/overview#large-language-models) have incredible generation capabilities,
 they struggle with discrete tasks (e.g. mathematics) and interacting with the outside world (e.g. getting the weather).
 
-Tools can be thought of as programs you give to a model which can be run as and when the model deems applicable.
+Tools are actions that an LLM can invoke.
+The results of these actions can be reported back to the LLM and it can consider them in the next response.
+
+For example, when you ask an LLM for the "weather in London", and there is a weather tool availalbe, it could call a tool
+with London as the argument. The tool would then fetch the weather data and return it to the LLM. The LLM can then use this
+information in its response.
 
 ## What is a tool?
 
 A tool is an object that can be called by the model to perform a specific task.
-You can use tools with functions across the AI SDK (like [`generateText`](/docs/reference/ai-sdk-core/generate-text) or [`streamUI`](/docs/reference/ai-sdk-rsc/stream-ui)) by passing a tool(s) to the `tools` parameter.
+You can use tools with functions across the AI SDK (like [`generateText`](/docs/reference/ai-sdk-core/generate-text)
+or [`streamUI`](/docs/reference/ai-sdk-rsc/stream-ui)) by passing one or more tools to the `tools` parameter.
 
-There are three elements of a tool, a description, parameters, and an optional `execute` or `generate` function (dependent on the SDK function).
+A tool consists of three properties:
 
 - **`description`**: An optional description of the tool that can influence when the tool is picked.
 - **`parameters`**: A [Zod schema](/docs/foundations/tools#schema-specification-and-validation-with-zod) or a [JSON schema](/docs/reference/ai-sdk-core/json-schema) that defines the parameters. The schema is consumed by the LLM, and also used to validate the LLM tool calls.
-- **`execute`** or **`generate`**: An optional async or generator function that is called with the arguments from the tool call.
+- **`execute`**: An optional async function that is called with the arguments from the tool call.
 
-## Tool Calls
+<Note>
+  `streamUI` uses UI generator tools with a `generate` function that can return
```

Comment:
rather than as a generator function, I would saying, using generator functions. Although this may not even be necessary on the foundations page.